src/Components/Sidebar/WorkspaceCustom.tsx

```diff
@@ -0,0 +1,122 @@
+// eslint-disable
+
+import React, { useEffect, useState } from 'react'
+import TreeView from '@material-ui/lab/TreeView'
+import TreeItem from '@material-ui/lab/TreeItem'
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
+import ChevronRightIcon from '@material-ui/icons/ChevronRight'
+import { FcGraduationCap, FcAssistant } from 'react-icons/fc'
+import './Workspace.scss'
+import { useQuery, gql } from '@apollo/client'
+import { Button } from '@material-ui/core'
+import { Link } from 'react-router-dom'
+import { makeStyles } from '@material-ui/core/styles'
+
+const useStyles = makeStyles((theme) => ({
+  container: { marginTop: theme.spacing(1) },
+  link: {
+    color: 'black',
+    textDecoration: 'none',
+  },
+}))
+
+const GET_WORKSPACES = gql`
+  query allWorkspaces($input: InputWorkspaceGet!) {
+    allWorkspaces(input: $input) {
+      id
+      title
+      isSchoolWorkspace
+      feed {
+        id
+        feedName
+        messages {
+          id
+          content
+          userId
+          likes
+          dislikes
+          comments {
+            id
+            content
+            userId
+            createdAt
+          }
+        }
+      }
+      assets {
+        id
+        assetName
+      }
+      visio
+    }
+  }
+`
+
+const WorkspaceCustom: React.FC = () => {
+  const { loading, error, data } = useQuery(GET_WORKSPACES, {
+    variables: {
+      input: {
+        isSchoolWorkspace: false,
+        schoolId: '2',
+      },
+    },
+  })
+  const [workspace, setWorkspace] = useState([])
+  const classes = useStyles()
+  useEffect(() => {
+    if (data) {
+      setWorkspace(data.allWorkspaces)
+    }
+  }, [data])
+
+  if (loading) return <p>Loading...</p>
+  if (error) return <p>Error :(</p>
+
+  return (
+    <div className={classes.container}>
+      {workspace ? (
+        <div className="workspace_container">
+          <div className="title_container">
+            <FcAssistant className="workspace_icon" />
+            <p>Espace de travail</p>
+          </div>
+          <TreeView
+            defaultCollapseIcon={<ExpandMoreIcon />}
+            defaultExpandIcon={<ChevronRightIcon />}
+            className="treeview"
+          >
+            {workspace.map((el: any) => (
```

Comment:
you can create a type or an interface for workspace to replace the any 
eg: 
type TWorkspace = {
id : numebr
title : string
isSchoolWorkspace : boolean
ect ... 
}
we can create a file with all the interfaces in the project afterwards so that it is less heavy in the component